### ISSUE TITLE
S3TransferUtilitySampleSwift returns "Failed with error: Error Domain=..." when downloading image

### ISSUE THREAD

**Issue by davidskun** [awslabs/aws-sdk-ios-samples#168]
I'm using iOS 10, Swift 3, Xcode 8.1 to run the S3TransferUtilitySampleSwift app.  I modified the portions of the info.plist, capabilities, and Constants.swift.  I know I have a valid Cognito pool ID and bucket since they work on the Android app.

However, when I try to download an image given the bucket and key, I get the following error:

`S3TransferUtilitySampleSwift[7256:152627]`` Failed with error: Error Domain=com.amazonaws.AWSS3TransferUtilityErrorDomain Code=2 "(null)" UserInfo={Server=AmazonS3, Transfer-Encoding=Identity, Connection=close, Content-Type=application/xml, Date=Sat, 04 Mar 2017 10:53:43 GMT, x-amz-request-id=3DFF780A2317BA95, x-amz-id-2=j0RqkR4jIZZ5Nk83lpmpgXhOWY0nOdr1be68Ki6qHPlqg+vO9Qvqkm7sybpz5kaBuvrJX9JLixM=}`

Any ideas on why this is happening?  Thanks in advance!

**Comment by rohandubal** [comment]
Hi,

Is your bucket region set correctly? This seems like an issue related to invalid configuration.

Thanks,
Rohan

**Comment by davidskun** [comment]
Hi Rohan,

I set the region in the Info.plist to USEast1 (as called out in the markdown) of the S3TransferUtilitySwift sample app.  I haven't tried other regions, but it's worth a shot.  I'll try USWest1.

Any other ideas?

Thanks,
Dave

**Comment by davidskun** [comment]
USWest1 didn't work.  I received the following error.

`Session task failed with error: Error Domain=NSURLErrorDomain Code=-1003 "A server with the specified hostname could not be found." `

**Comment by rohandubal** [comment]
The sample app tries to download the image which is uploaded from the upload tab. Could you please upload the image first and then try the download? Make sure that the image / key name being downloaded is present in the bucket.

Thanks,
Rohan

**Comment by davidskun** [comment]
I changed the Constants.swift to point to my bucket and key name of an existing image in that bucket.  But get the same error while downloading.  Not sure what else to try.

**Comment by mlshon** [comment]
In what region did you create your bucket?

**Comment by medampudi** [comment]
has any one found the solution for this issue.. I dont think my credentials have the right management  for Accessing the S3. can you please point me post to follow thru. I am getting the following example

`Optional(Error Domain=com.amazonaws.AWSS3TransferUtilityErrorDomain Code=2 "(null)" UserInfo={Server=AmazonS3, Transfer-Encoding=Identity, Connection=close, Content-Type=application/xml, Date=Wed, 05 Apr 2017 16:04:26 GMT, x-amz-request-id=0C93CF95071C9490, x-amz-id-2=j4Vu/iWVMuxTcwLNxTpqQd9CSH4OUyJ9GC92d/E+V4MOmHn4dsjqME6JZZ1Vp6E6oCq0mf59MU8=})`

**Comment by bdel23** [comment]
I'm having the same issue. 

S3TransferUtilitySampleSwift[22621:13754909] Failed with error: Error Domain=com.amazonaws.AWSS3TransferUtilityErrorDomain Code=2 "(null)" UserInfo={x-amz-id-2=vRjKMnOa21ry0xF/DKQpz57zKONmlQkaz916Elfwvs7CzLmSknjdM2BWwEQQHX1WL/Aj63KzHKw=, Server=AmazonS3, Transfer-Encoding=Identity, x-amz-request-id=5FB3000E60EE3B00, Content-Type=application/xml, Date=Sat, 01 Jul 2017 22:35:45 GMT}

I have checked my permissions and region names, but cannot get this to work.  I have spent a couple days on this and I'm out of ideas.  Anyone figure this out?
